src/vmstats_server.erl

```diff
@@ -22,14 +29,19 @@
                 prev_gc :: {GCs::integer(), Words::integer()}}).
 %%% INTERFACE
 start_link(Sink, BaseKey) ->
-    gen_server:start_link({local, ?MODULE}, ?MODULE, {Sink, BaseKey}, []).
+    start_link(Sink, BaseKey, []).
+
+start_link(Sink, BaseKey, Options) ->
+    gen_server:start_link({local, ?MODULE}, ?MODULE, {Sink, BaseKey, Options}, []).
 
 %%% INTERNAL EXPORTS
-init({Sink, BaseKey}) ->
-    {ok, Interval} = application:get_env(vmstats, interval),
-    {ok, KeySeparator} = application:get_env(vmstats, key_separator),
-    {ok, SchedTimeEnabled} = application:get_env(vmstats, sched_time),
-    {ok, MemoryMetrics} = application:get_env(vmstats, memory_metrics),
+init({Sink, BaseKey, UserOptions}) ->
+    Options = UserOptions ++ application:get_all_env(vmstats),
+    Interval = proplists:get_value(interval, Options, 1000),
+    KeySeparator = proplists:get_value(key_separator, Options, $.),
+    SchedTimeEnabled = proplists:get_value(sched_time, Options, true),
+    MemoryMetrics = proplists:get_value(memory_metrics, Options,
```

Comment:
I'd use lists:keyfind with a hard pattern match -- the default values are provided by the .app's env tuple (as you could see before, no defaults were given in code).

Otherwise this implicitly creates a new layer of default options in the code that backstops the .app file, but we shouldn't introduce that unless we really need to